LibertyDSNP/frequency
# feat: Update dependencies and toolchain versions

# Goal
The goal of this PR is to update the rust toolchain to the `stable` channel and publish a new version of `ci-base-image`. 
GitHub workflows that use `ci-base-image` cannot be updated to use `nightly-2024-02-04` until a new version of `ci-base-image` is published (see Follow-up)

Closes #1854

# Changes
- `rust-toolchain.toml` updated channel to be `stable`
- `Makefile` targets updated to use `nightly-2024-02-04` instead of `nightly-2023-07-13`
  - `lint` no longer needs `nightly` unstable features for clippy
- `ci-base-image.dockerfile` use `nightly-2024-02-04` instead of `nightly-2023-07-13`
  - Add `RUST_VERSION` to keep track of the rust toolchain
- `merge-pr.yml` update `ci-base-image` version to `1.2.0` and add `RUST_VERSION`


# How to Test
- Make sure all tests pass. [There are no code changes, only toolchain and workflow updates]
- Test `make format` and `make lint` which use `nightly`

# Follow-up Issue #1896 
- GitHub workflows should be updated to use `nightly-2024-02-04` and `RUST_TOOLCHAIN` should be updated to `stable` once a new `ci-base-image` is published
- `Cargo.lock` may be updated by `cargo update`

## DEVELOPER NOTE: Check your `rustc --version` and make sure you are on the latest (1.76) `rustup update` to get the latest.

# Checklist
- [ ] Weights updated
